CMakeLists.txt

```diff
@@ -70,12 +76,26 @@ if(USE_ASAN)
   link_libraries(asan)
 endif()
 
+# Enable testing if we are building tests
+if(BUILD_TESTS)
+  enable_testing()
+endif(BUILD_TESTS)
+
 # Output the compilation database
 set(CMAKE_EXPORT_COMPILE_COMMANDS
     ON
     CACHE STRING "Enable/Disable output of compile commands during generation." FORCE
 )
 
+# Enable all warnings that we can
+add_compile_options(-Wall -Wpedantic -Wextra)
+
+# This is needed everywhere so that we can catch segmentation faults
+add_compile_options(-fnon-call-exceptions)
```

Comment:
You've removed `-Wno-int-in-bool-context`. Is that warning gone now?

Reply:
This warning was fixed by the modern cmake stuff. The reason it showed up was that the Eigen library we used had that warning and it wasn't properly being included as SYSTEM. Which meant our compiler treated it as "our" codes issue not one in an included library. Now that it's properly set we can have that rule again, and if we break it in our code we should just not use an int in a bool context